### PR TITLE
feat: add show/hide visibility feature for total loan and saving amounts

### LINKED ISSUE #2003: UI Enhancement for Privacy: Feature to Show/Hide Savings and Loan Amounts on Dashboard.
**Summary:**

A toggle button on the app's dashboard enables the user to display or hide loan and savings amounts. This feature addresses user privacy and cybersecurity concerns and has become a standard practice in banking and financial applications.

**Steps to reproduce:**

 - Open the app
 - See total savings and loan amounts under the account overview section

**Expected behaviour:**

In line with modern UI, the app lets users choose whether to display or hide financial information.

**Observed behaviour:**

In the home screen both the saving and loan amounts are visible directly without requiring any additional action from the user and this could potentially impact the user experience.

**Device and Android version:**

Device : Google Pixel 4a
Android Version: 13

**Screenshots:**
photo_2023-03-17_06-46-23

### PR DESCRIPTION
**Pull request by lssarao**
Fixes #2003

**Summary**
Added a single-tap icon button to the app's dashboard that enables the user to display or hide loan and savings amounts by masking them.

**Screenshots**
</img> </img> 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by lssarao** [comment]
@jawidMuhammadi, please review this PR. I have thoroughly tested my changes to meet project requirements and coding standards. Please let me know if you have any suggestions on this, Thank You.

**Comment by chayan-dev** [comment]
@lssarao any specific reason to add onViewCreated() in HomeOldFragment and add new code in it?

**Comment by lssarao** [comment]
@chayan-dev I used onViewCreated() function to initialize views and set up event listeners due to the function's invocation once the fragment's view hierarchy is created, resulting in better performance and making it a suitable place to add new code.

**Comment by lssarao** [comment]
@jawidMuhammadi It seems that the workflow requires your approval before it can be executed. Would you kindly review and approve it so that we can proceed with the PR

**Comment by lssarao** [comment]
@jawidMuhammadi I'm following up on this, the PR has passed all workflows, so please merge it into the main branch. If there are any issues with the code, please let me know, and I will address them promptly. Thank you for your time.

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]
In other parts of the project we use ButterKnife, let's use that library instead of `findViewById`.

**Comment by lssarao** [comment]
@jawidMuhammadi I have bound the views with ButterKnife. Could you please review and merge this. Thank you for your time and guidance.

File: app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt]: Remove this comment, please and extract all the toggle-related code to a sperate function and give that function an explanatory name.

File: app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt]: You can set the initial value of visibility to GONE in the XML file. The same applies to the codes in line number 280 to 283

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt
Comment (lssarao) [reply app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt]: @jawidMuhammadi I've removed the comment and moved the toggle-related code to a separate function named toggleVisibilityButton.

File: app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt
Comment (lssarao) [reply app/src/main/java/org/mifos/mobile/ui/fragments/HomeOldFragment.kt]: @jawidMuhammadi  I've set the initial visibility of the view using the android visibility attribute in XML.

**Comment by lssarao** [comment]
@jawidMuhammadi could you please review it now and merge this if you are satisfied, Thanks for your feedback.

**Review by jawidMuhammadi — APPROVED** [review]
Nice Job! 👏

**Pull request merged**
